pkg/util/cmdutil/flag.go

```diff
@@ -34,3 +40,53 @@ func IncompatibleFlags(cmd *cobra.Command, first, second string) error {
 	}
 	return nil
 }
+
+// DecodeFlags decodes the set flags of a cobra.Command and unpacks all the
+// values to the specified pointer of the passed structure.
+func DecodeFlags(cmd *cobra.Command, dst interface{}) error {
+	var flagMap = make(map[string]interface{})
+	cmd.Flags().VisitAll(func(f *pflag.Flag) {
+		flagMap[f.Name] = parseValue(f.Value)
+	})
+
+	return mapstructure.WeakDecode(flagMap, dst)
+}
+
+func parseValue(val pflag.Value) interface{} {
+	// All types which encapsulate the pflag slice type have a GetSlice
+	// method, which obtains the slice as []string. Combined with the
+	// mapstructure call to WeakDecode, it'll decode the slice correctly.
+	t := val.Type()
+	sliceOrArray := strings.HasSuffix(t, "Slice") || strings.HasSuffix(t, "Array")
```

Comment:
can we extract this into a separate `isSliceOrArrayFlag()` function ? I think it would improve readability

Reply:
Pretty much all of the `parseValue` function is geared towards parsing the slice type. I've added better comments and var names, now the function should be a bit clearer and more readable.

Reply:
thank you!